src/tests/ftest/control/dmg_storage_query.py

```diff
@@ -58,18 +98,44 @@ def test_dmg_storage_query_devices(self):
         :avocado: tags=DmgStorageQuery,test_dmg_storage_query_devices
         """
         # Get the storage device information, parse and check devices info
+        self.log_step('Get the storage device information')
         device_info = get_storage_query_device_info(self, self.dmg)
 
         # Check if the number of devices match the config
-        msg = "Number of devs do not match cfg: {}".format(len(self.bdev_list))
-        self.assertEqual(len(self.bdev_list), len(device_info), msg)
+        self.log_step('Verify storage device count')
+        if len(self.bdev_list) != len(device_info):
+            self.fail(
+                'Number of devices ({}) do not match server config ({})'.format(
+                    len(device_info), len(self.bdev_list)))
 
         # Check that number of targets match the config
+        self.log_step('Verify storage device targets and roles')
+        errors = 0
         targets = 0
         for device in device_info:
+            self.log.info('Verifying device %s', device['tr_addr'])
             targets += len(device['tgt_ids'])
+            self.log.info('  targets: detected=%s', len(device['tgt_ids']))
+            message = ''
+            for bdev in self.bdev_list:
+                bdev_tr_addr = '{:02x}{:02x}{:02x}:'.format(
+                    *list(map(int, re.split(r'[:.]', bdev['bdev'])[1:], [16] * 3)))
```

Comment:
Just due to the complexity of these two lines, does this warrant a function with documentation?

Reply:
Added a comment